### ISSUE TITLE
Final accuracy of GPT2 (1.5 B) using openWebText on Megatron-LM 

### ISSUE THREAD

**Issue by codecaution** [NVIDIA/Megatron-LM#102]
Hi.
I try to reproduce the experiment on GPT2(1.5B) with Megatron-LM recently, but I don't get the expected accuracy performance.
I conduct the experiments by following the steps below.
1. Following the instruction, I collect GPT Webtext Data using the default argument in [download.py](https://github.com/yet-another-account/openwebtext/blob/master/download.py) and process the data as [here](https://github.com/NVIDIA/Megatron-LM/tree/main/tools/openwebtext).
2. Pretrain GPT2 with 4xDGX-A100 (32GPU) and configs as follow:
`python -m torch.distributed.launch $DISTRIBUTED_ARGS \
       pretrain_gpt.py \
       --num-layers 48 \
       --hidden-size 1600 \
       --num-attention-heads 25 \
       --micro-batch-size 8 \
       --global-batch-size 512 \
       --seq-length 1024 \
       --max-position-embeddings 1024 \
       --train-iters 1000000 \
       --lr-decay-iters 50000 \
       --save $CHECKPOINT_PATH \
       --load $CHECKPOINT_PATH \
       --data-path $DATA_PATH \
       --vocab-file $VOCAB_PATH \
       --merge-file $MERGE_PATH \
       --data-impl mmap \
       --split 949,50,1 \
       --distributed-backend nccl \
       --lr 0.00025 \
       --lr-decay-style cosine \
       --min-lr 1.0e-5 \
       --weight-decay 1e-2 \
       --clip-grad 1.0 \
       --lr-warmup-iters 2000\
       --checkpoint-activations \
       --log-interval 100 \
       --save-interval 10000 \
       --eval-interval 1000 \
       --eval-iters 10 \
       --fp16 | tee $LOG_PATH
`
3. Evaluate the zero-shot performance on downstream tests, such as LAMBADA.
However, I find the final performance on LAMBADA is not good as expected, which reports 
> validation results on LAMBADA | number correct: 2.3330E+03 | total examples: 5.1530E+03 | avg accuracy: 4.5275E-01

I wonder if I have any wrong steps？

After comparing the [GPT2 paper](http://www.persagen.com/files/misc/radford2019language.pdf) and [Megatron-LM](https://arxiv.org/abs/1909.08053) paper, I have a question that **Why Megatron-LM trained on bigger dataset(collection of Wikipedia, CC-Stories, RealNews, and OpenWebtext, ~174GB) but get lower accuracy(45.18% on LAMBADA) than openAI GPT2 (55.48% on LAMBADA) trained only on OpenWebtext(~40GB) with similar parameter size.**  

Could you give some help? Thank you very much.

**Comment by lucasleesw** [comment]
hi, could you share what the results of [download.py ](https://github.com/yet-another-account/openwebtext/blob/master/download.py) look like? After downloading process, I found nothing in the data folder. Thank you so much :)
